Tree refactor for expansion

svouse opened the pull request:
moved tree functions, removed extra src import in index

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly-devtools/40)
<!-- Reviewable:end -->

Commit:
Fixed tree bug

Commit:
Moving tabs in prep for workspace reconfigure

Commit:
amending for CLA. comments flagging deletion issue

Commit:
uncommented flagged lines

Commit:
removed flag comments

Commit:
removed nw.js console command

Commit:
removed extra edits

Commit:
removed unnecessary line

Commit:
fixed spacing

Commit:
uncommented commented code

Commit:
more spacing

Commit:
indents

Commit:
newline

Commit:
readded deleted code

Commit:
more spacing

Commit:
comment punctuation

Commit:
newline

Commit:
newline

Commit:
added TODO in wfactory controller

Commit:
removed extra call to showStarterBlock

Commit:
fixed indentation and issue #24

Commit:
removed unnecessary setSelectedBlockType

Commit:
removed call to nw.js devtools

Commit:
fixed clear library issue

Commit:
removed lib assignment; listener uses lambda

Commit:
changed TODO styling

Commit:
Merge branch 'develop' of https://github.com/google/blockly-devtools into delete_debug

Commit:
readded ES6 formatting

Commit:
changed event listener

Commit:
Merge branch 'develop' of https://github.com/google/blockly-devtools into feature_collapseTree

Commit:
Merge branch 'delete_debug' into feature_collapseTree

Commit:
Moved tree management to project_controller

Commit:
removed extra jquery inclusion, moving tree fctns.

Commit:
readding constructor

Commit:
fixed all known bugs

Commit:
newline removal

Commit:
removed nw.js debug

AnmAtAnm commented on src/app_view.js:
```diff
@@ -71,19 +71,28 @@ class AppView {
     // Dictionary with all MenuItems. Keys are the labels, values are the
     // nw.MenuItem's. Values are added in initMenuTree().
-    this.menuItems = {};
+    this.menuItems = {}
```
Google's JavaScript style guide demands semicolons .  Here and few lines below.

AnmAtAnm commented on src/app_view.js:
```diff
@@ -71,19 +71,28 @@ class AppView {
   }
 
   /**
+  * Initializes the tree for the session.
```
I'm not sure there is a written rule about this, but each line of a /* or /** comment is usually aligned against the first * of that line.  In other words, in this case, one more space of indent for each line after the first.

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+
+/**
+ * NavigationTree Class
+ * @param {!BlockLibraryController) libraryController allows tree to get blocks
```
Mismatched { and ).  I think you mean }

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+ * @param {!BlockLibraryController) libraryController allows tree to get blocks
+  * in library
+ * @constructor
+ */
```
Correct indent for the above two lines

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+    /**
+   * Returns JSON object of library's blocktypes.
+   * @return JSON of all block types
+   */
```
Again about /** indent alignment.

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+
+    /**
+   * Returns JSON object of library's blocktypes.
+   * @return JSON of all block types
```
Missing type {!Object}

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+  constructor(libraryController){
+   this.libraryController = libraryController;
+   var blocks = this.libraryController.getStoredBlockTypes();
+  this.buildTree(blocks);
```
Correct indent. Two spaces more than constructor above.

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+  * in library
+ * @constructor
+ */
+  constructor(libraryController){
```
Space before {

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+ * @author sagev (Sage Vouse)
+ */
+
+class NavigationTree{
```
Space before {

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+    // TODO(#27) : upon giving libraries names add them as roots under the project
+    var data = this.makeBlockTypeJson();
+    var tree = {
+      "core" : {
```
No space before :

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+            "label": "Add",
+            "action": function (obj) {
+              $('#navigationTree').jstree().create_node('#' , { "id" :
+                "ajason5", "text" : "new_block"}, "last", null);
```
Line break at the "prefer to break at a higher syntactic level" [link](https://google.github.io/styleguide/jsguide.html#formatting-where-to-break).
In this case, don't break in the middle of a key/value pair. Prefer putting one key/value pair on each line if they don't all fit on the same line.

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+    var blockType;
+    while (types[iterationIndex]) {
+      blockType= types[iterationIndex - 1];
+      toAdd = {"text" : blockType, "id" : blockType};
```
Google style dictates single quotes for strings ([link](https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes)).  The only exception is JSON that is intended to be extracted from the JavaScript source, since JSON does not allow single quote strings.

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+    $('#navigationTree').jstree().create_node('#' , {"id" : blockType,
+      "text" : blockType }, "last", null);
+  }
+}
```
Nit: Additional newline for consistency

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+  }
+
+  clearLibrary(){
+    // TODO(#25): make more elegant
```
Does this still apply?

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+    });
+  }
+
+  clearLibrary(){
```
Space before {
Add doc comment and methods below.

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+  clearLibrary(){
+    // TODO(#25): make more elegant
+    // clear the tree
+    $('#navigationTree').jstree("destroy");
```
You managed to use both single quote and double quote on the same line. Almost always use single quote.

AnmAtAnm commented on src/navigation_tree.js:
```diff
@@ -0,0 +1,145 @@
+  }
+
+  addBlockNode(blockType){
+    $('#navigationTree').jstree().create_node('#' , {"id" : blockType,
```
Remove space between `'#'` and `,`.
The Object literal can fit on one line (although one line per key would probably be more readable). In fact, the whole parameter sequence might, if you want.
Use single quotes.

AnmAtAnm commented:
I looks like these commits were done via your `@google` account, again. Check the git config of your local repo.

Also, I highly recommend setting up `eslint` on your machine and running it before each commit.

Commit:
fixed missing param

Commit:
fixed missing param, changed commit email

Commit:
CLA ammend

svouse committed:
fixed spacing

Commit:
fixed semicolon

Commit:
style fixes

Commit:
fixing style mistakes, adding forgotten comments

Commit:
':' space

Commit:
" to '

Commit:
" to '

Commit:
" to '

Commit:
Merge branch 'expanded_tree' of https://github.com/google/blockly-devtools into expanded_tree

Commit:
fixed indent

svouse commented:
<comment to get bot to rerun>

svouse commented:
Bot rerun comment

svouse commented:
bot rerun comment

svouse commented:
bot rerun

AnmAtAnm commented:
Reviewed 2 of 7 files at r2, 1 of 4 files at r4.
Review status: 3 of 6 files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[package.json, line 10 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmfsGbCJjhOtmWZioD:-KmmfsGbCJjhOtmWZioE:bqu74n) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/package.json#L10)):*
> ```JSON
>   "dependencies": {
>     "component-emitter": "github:component/emitter",
>     "eslint": "^4.0.0",
> ```

This should be in `devDependencies`, because we only need it during develop time.
https://docs.npmjs.com/files/package.json#devdependencies

---

*[src/app_view.js, line 77 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgJ789RG-eKNhyDo2:-KmmgJ789RG-eKNhyDo3:b-qbf8ac) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/app_view.js#L77)):*
> ```JavaScript
> 
>     // Initializes menubar.
>     this.mainMenu = new nw.Menu({type: 'menubar'})
> ```

semicolon

---

*[src/app_view.js, line 78 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgKL7Yjn-rBB-PcBU:-KmmgKL7Yjn-rBB-PcBV:b-qbf8ac) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/app_view.js#L78)):*
> ```JavaScript
>     // Initializes menubar.
>     this.mainMenu = new nw.Menu({type: 'menubar'})
>     this.initMenuTree(this.mainMenu, this.menuTree)
> ```

semicolon

---

*[src/app_view.js, line 253 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgN0f9ZuhX62ND1of:-KmmgN0f9ZuhX62ND1og:b-yodq3a) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/app_view.js#L253)):*
> ```JavaScript
> }
> ```

Nit: Extra newline

---

*[src/block_library_controller.js, line 66 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgScawN033VNvEQi-:-KmmgScawN033VNvEQi0:bmdjz95) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/block_library_controller.js#L66)):*
> ```JavaScript
> * @param {!JSTree} tree the main navigation tree
> */
> BlockLibraryController.prototype.setTree = function(tree){
> ```

space before {

---

*[src/navigation_tree.js, line 82 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-Kmmgv7eEskrxW71B6lq:-Kmmgv7eEskrxW71B6lr:b-q3dit2) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/navigation_tree.js#L82)):*
> ```JavaScript
>         'data': data
>       },
>       'plugins' : [ 'contextmenu', 'dnd', 'crrm'],
> ```

No space before the `:`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/40)*
<!-- Sent from Reviewable.io -->

Commit:
fixed spacing/added missing semicolons

svouse commented:
bot rerun comment

svouse committed:
style fixes

svouse commented:
Review status: 2 of 8 files reviewed at latest revision, 23 unresolved discussions.

---

*[package.json, line 10 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmfsGbCJjhOtmWZioD:-Kmn-vQilU4Nz8X-dNi4:bs7wumb) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/package.json#L10)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

This should be in `devDependencies`, because we only need it during develop time.
https://docs.npmjs.com/files/package.json#devdependencies

</blockquote></details>

Done.

---

*[src/app_view.js, line 74 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE3-r2-74:-Kmn02JiVv-cq7rLqKM-:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/app_view.js#L74)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Google's JavaScript style guide demands semicolons .  Here and few lines below.
</blockquote></details>

Done.

---

*[src/app_view.js, line 83 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE3-r2-83:-Kmn0WEEbdctP4em3a0A:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/app_view.js#L83)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

I'm not sure there is a written rule about this, but each line of a /* or /** comment is usually aligned against the first * of that line.  In other words, in this case, one more space of indent for each line after the first.
</blockquote></details>

Done.

---

*[src/app_view.js, line 77 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgJ789RG-eKNhyDo2:-Kmn03GL-n2eRvyG53TL:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/app_view.js#L77)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

semicolon
</blockquote></details>

Done.

---

*[src/app_view.js, line 78 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgKL7Yjn-rBB-PcBU:-Kmn04MPDbPOfJtOT7i2:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/app_view.js#L78)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

semicolon
</blockquote></details>

Done.

---

*[src/app_view.js, line 253 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgN0f9ZuhX62ND1of:-Kmn0Z57ehV93ZseVlCk:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/app_view.js#L253)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Nit: Extra newline
</blockquote></details>

Done.

---

*[src/block_library_controller.js, line 66 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgScawN033VNvEQi-:-Kmn0gw0VIEwGUslk3ib:bddwnk6) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/block_library_controller.js#L66)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

space before {
</blockquote></details>

Done; also fixed comment indent

---

*[src/navigation_tree.js, line 28 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-28:-Kmn0mZ99FC0HkgVRKbg:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L28)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Space before {
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 32 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-32:-Kmn0nYa1eml55ZXiC__:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L32)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Mismatched { and ).  I think you mean }
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 35 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-35:-Kmn1AY3GOhizCxByUzI:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L35)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Correct indent for the above two lines
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 36 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-36:-Kmn1CH_cx8tEO--b3Dp:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L36)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Space before {
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 39 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-39:-Kmn1F159iPWAOQPXi2i:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L39)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Correct indent. Two spaces more than constructor above.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 44 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-44:-Kmn1P_t8T5VK6HrHOgv:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L44)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Missing type {!Object}
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 45 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-45:-Kmn1R7YlNlwfIFtZqpc:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L45)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Again about /** indent alignment.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 58 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-58:-Kmn1fHo8XpDrYIX4LTD:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L58)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Google style dictates single quotes for strings ([link](https://google.github.io/styleguide/jsguide.html#features-strings-use-single-quotes)).  The only exception is JSON that is intended to be extracted from the JavaScript source, since JSON does not allow single quote strings.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 78 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-78:-Kmn1gNN-p8JWGo1jbZU:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L78)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

No space before :
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 89 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-89:-Kmn1x16CQHqYdH0za48:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L89)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Line break at the "prefer to break at a higher syntactic level" [link](https://google.github.io/styleguide/jsguide.html#formatting-where-to-break).
In this case, don't break in the middle of a key/value pair. Prefer putting one key/value pair on each line if they don't all fit on the same line.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 128 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-128:-Kmn1zpKFt3R6iSHZUd6:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L128)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Space before {
Add doc comment and methods below.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 129 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-129:-Kmn20U3jIWQz6o0h9If:b-h8m1dh) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L129)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Does this still apply?
</blockquote></details>

Nope; removed it

---

*[src/navigation_tree.js, line 131 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-131:-Kmn26guBkVen4YSbcz9:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L131)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

You managed to use both single quote and double quote on the same line. Almost always use single quote.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 142 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-142:-Kmn3Ai-6UOXqGbdyuSd:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L142)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Remove space between `'#'` and `,`.
The Object literal can fit on one line (although one line per key would probably be more readable). In fact, the whole parameter sequence might, if you want.
Use single quotes.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 145 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-145:-Kmn3QX0G8DXhBwr6g70:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L145)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Nit: Additional newline for consistency
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 82 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-Kmmgv7eEskrxW71B6lq:-Kmn1gwnlw6leBQfNATk:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/navigation_tree.js#L82)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

No space before the `:`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/40)*
<!-- Sent from Reviewable.io -->

svouse committed:
removed unnecessary config change

svouse committed:
Merge branch 'expanded_tree' of https://github.com/google/blockly-devtools into expanded_tree

svouse committed:
removed unncessary addition

svouse committed:
fixed comment level that had undone in rebase

AnmAtAnm commented:
Reviewed 2 of 5 files at r5.
Review status: 4 of 7 files reviewed at latest revision, 5 unresolved discussions.

---

*[package.json, line 13 at r5](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmnFn-NnLRFR50HmGYf:-KmnFn-NnLRFR50HmGYg:b-xq9n0c) ([raw file](https://github.com/google/blockly-devtools/blob/4ec49136802c6e23dc76d07d35fad6b4702cb329/package.json#L13)):*
> ```JSON
>   },
>   "devDependencies": {
>     "component-emitter": "github:component/emitter",
> ```

I only meant eslint. emitter is a regular dependency. It's used in the app to manage listeners in the view/controller framework.
Move it back up.

---

*[src/block_library_controller.js, line 66 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgScawN033VNvEQi-:-KmnGUMUzUWBgePcFXuL:bxphd49) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/block_library_controller.js#L66)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Done; also fixed comment indent
</blockquote></details>

* alignment in comment block

---

*[src/navigation_tree.js, line 142 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-142:-KmnHnLuQhgXpaPv9Qlg:bifvr19) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L142)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Object literal is still split on two lines.

---

*[src/navigation_tree.js, line 28 at r5](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmnGkAVKl8SoWN8LVgW:-KmnGkAVKl8SoWN8LVgX:b-iydzui) ([raw file](https://github.com/google/blockly-devtools/blob/4ec49136802c6e23dc76d07d35fad6b4702cb329/src/navigation_tree.js#L28)):*
> ```JavaScript
>  */
> 
> class NavigationTree {
> ```

The class should have a JSDoc comment.

---

*[src/navigation_tree.js, line 111 at r5](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmnHCQe6HK29O31qEhU:-KmnHCQe6HK29O31qEhV:b-ga6tb9) ([raw file](https://github.com/google/blockly-devtools/blob/4ec49136802c6e23dc76d07d35fad6b4702cb329/src/navigation_tree.js#L111)):*
> ```JavaScript
>    * Populates the tree and adds its listener.
>    */
>   buildTree(blockTypes) {
> ```

@param for blockTypes

Also, is there a thought process behind this method being `build` and all the others being `make`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/40)*
<!-- Sent from Reviewable.io -->

svouse committed:
style fixes

svouse committed:
fixed object literal

svouse commented:
Review status: 4 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

*[package.json, line 13 at r5](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmnFn-NnLRFR50HmGYf:-KmnJb4sgJB87p5ZrhaD:bg4rf6e) ([raw file](https://github.com/google/blockly-devtools/blob/4ec49136802c6e23dc76d07d35fad6b4702cb329/package.json#L13)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

I only meant eslint. emitter is a regular dependency. It's used in the app to manage listeners in the view/controller framework.
Move it back up.
</blockquote></details>

Whoops; fixed

---

*[src/navigation_tree.js, line 142 at r2](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmigcoY7um3Elg8ynE4-r2-142:-KmnKjsN7KHkoM90UEUG:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/69ec625b78818473deffcbe89b12fe84da10699b/src/navigation_tree.js#L142)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Object literal is still split on two lines.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 28 at r5](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmnGkAVKl8SoWN8LVgW:-KmnKqdM9f5j9KgmKIPr:b-896fix) ([raw file](https://github.com/google/blockly-devtools/blob/4ec49136802c6e23dc76d07d35fad6b4702cb329/src/navigation_tree.js#L28)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

The class should have a JSDoc comment.
</blockquote></details>

Done.

---

*[src/navigation_tree.js, line 111 at r5](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmnHCQe6HK29O31qEhU:-KmnK8cTVKNS2nTrf_FT:btgufit) ([raw file](https://github.com/google/blockly-devtools/blob/4ec49136802c6e23dc76d07d35fad6b4702cb329/src/navigation_tree.js#L111)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

@param for blockTypes

Also, is there a thought process behind this method being `build` and all the others being `make`?
</blockquote></details>

nope; fixed for consistency

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/40)*
<!-- Sent from Reviewable.io -->

svouse commented:
Review status: 4 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/block_library_controller.js, line 66 at r4](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmmgScawN033VNvEQi-:-KmnKtYA5rS4x7Achrdn:b-lvrt8i) ([raw file](https://github.com/google/blockly-devtools/blob/6bcc7abc3a4adeba5f6abb6823f1c7e06349733a/src/block_library_controller.js#L66)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

* alignment in comment block
</blockquote></details>

Done. (sorry; had accidentally undone in rebase)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/40)*
<!-- Sent from Reviewable.io -->

AnmAtAnm commented:
Reviewed 2 of 3 files at r6.
Review status: 6 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/navigation_tree.js, line 111 at r5](https://reviewable.io:443/reviews/google/blockly-devtools/40#-KmnHCQe6HK29O31qEhU:-KmnKrsSXu3luCTEYYsR:b7stgbk) ([raw file](https://github.com/google/blockly-devtools/blob/4ec49136802c6e23dc76d07d35fad6b4702cb329/src/navigation_tree.js#L111)):*
<details><summary><i>Previously, svouse wrote…</i></summary><blockquote>

nope; fixed for consistency
</blockquote></details>

Still need `@param`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/40)*
<!-- Sent from Reviewable.io -->

svouse committed:
removed param

svouse committed:
spacing on '='

AnmAtAnm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 6 of 7 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-devtools/40#-:-KmnMVn6KwbeluBw4LLd:bnfp4nl)*
<!-- Sent from Reviewable.io -->
